spyder/plugins/explorer/tests/test_explorer.py

```diff
@@ -46,5 +66,99 @@ def test_project_explorer(project_explorer):
     assert project_explorer
 
 
+project_dir = osp.join(
+        getcwd_or_home(),
+        'temp_dir_test_file_explorer_functions_copy_save_path_file')
+if not osp.exists(project_dir):
+        os.mkdir(project_dir)
+project_file1 = osp.join(project_dir, 'script.py')
+with open(project_file1, 'w') as fh:
+    fh.write('Spyder4 will be released this year')
+project_file2 = osp.join(project_dir, 'pyscript.py')
+with open(project_file2, 'w') as fh:
+    fh.write('Spyder4')
+subdir = osp.join(project_dir, 'subdir')
+if not osp.exists(subdir):
+    os.mkdir(subdir)
+project_file3 = osp.join(subdir, 'Columbia.txt')
+with open(project_file3, 'w') as fh:
+    fh.write('South America')
+cb = QApplication.clipboard()
+
+
+@pytest.mark.parametrize('path_method', ['absolute', 'relative'])
+@pytest.mark.parametrize('file_paths', [[project_file1],
+                                        [project_file1, project_file2, subdir],
+                                        [project_file1, project_file3]])
+def test_copy_path(copy_path_file, path_method, file_paths):
+    """Test copy absolute and relative paths."""
+    project = copy_path_file
+    project.explorer.treewidget.copy_path(fnames=file_paths,
+                                          method=path_method)
+    cb_output = cb.text(mode=cb.Clipboard)
+    file_paths = [_fn.replace(os.sep, "/") for _fn in file_paths]
+    if len(file_paths) > 1:
+        if path_method == 'absolute':
+            true_path = ''.join('"' + _fn + '",' + '\n' for _fn in file_paths)
+        elif path_method == 'relative':
+            true_path = ''.join('"' + osp.relpath(_fn, getcwd_or_home()).
+                                replace(os.sep, "/") + '",' +
+                                '\n' for _fn in file_paths)
+        true_path = true_path[:-2]
+    else:
+        if path_method == 'absolute':
+            true_path = file_paths[0]
+        elif path_method == 'relative':
+            true_path = (osp.relpath(file_paths[0], getcwd_or_home()).
+                         replace(os.sep, "/"))
+    assert true_path == cb_output
+
+
+@pytest.mark.parametrize('file_paths', [[project_file1],
+                                        [project_file1, project_file2, subdir],
+                                        [project_file1, project_file3]])
+def test_copy_file(copy_path_file, file_paths):
+    """Test copy file(s)/folders(s) to clipboard."""
+    project = copy_path_file
+    project.explorer.treewidget.copy_file_clipboard(fnames=file_paths)
+    cb_data = cb.mimeData().urls()
+    for url in cb_data:
+        file_name = url.toLocalFile()
+        assert osp.isdir(file_name) or osp.isfile(file_name)
+        if file_paths == [project_file1]:
```

Comment:
Instead of hardcoding this for each paramaterizied scenario (which defeats much of the benefit of paramaterization), you should instead rely on checking that the length of ``url`` matches the length of the expected paths returned from your fixture after the modifications discussed above, that each file's path matches one in that list, and that its contents matches what is expected from the procedural generation above.

Reply:
Could you please revise again?, since I made some changes

Reply:
Good work so far; much of the remaining problems and complexity is due to not making the file content procedural generated from the filename and some static text as I requested above, which would allow you to drop all the special cases for each file is the main factor making things so messy. I'll address that in a separate review.

Reply:
Can we consider this as being resolved!